src/index.ts

```diff
@@ -209,17 +202,26 @@ export const derive = async function (privateKeyA: Buffer, publicKeyB: Buffer):
   if (publicKeyB.length === 33) {
     assert(publicKeyB[0] === 2 || publicKeyB[0] === 3, "Bad public key");
   }
-  const keyA = ec.keyFromPrivate(privateKeyA);
-  const keyB = ec.keyFromPublic(publicKeyB);
-  const Px = keyA.derive(keyB.getPublic()); // BN instance
-  return Buffer.from(Px.toArray());
+
+  // unpad to match previous implementation
+  // elliptic return BN and we return Buffer(BN.toArray())
+  // match by unpadding
+  const sharedSecret = secp256k1.getSharedSecret(privateKeyA, publicKeyB);
+  const Px = sharedSecret.subarray(sharedSecret.length - 32);
+
+  let i = 0;
+  while (i < Px.length && Px[i] === 0) {
+    i++;
+  }
+
+  return Px.subarray(i);
 };
 
 export const deriveUnpadded = derive;
 
-export const derivePadded = async function (privateKeyA: Buffer, publicKeyB: Buffer): Promise<Buffer> {
-  assert(Buffer.isBuffer(privateKeyA), "Bad private key");
-  assert(Buffer.isBuffer(publicKeyB), "Bad public key");
+export const derivePadded = async function (privateKeyA: Uint8Array, publicKeyB: Uint8Array): Promise<Uint8Array> {
```

Comment:
this does the same as `derive` except for the last part (padding)? if so, `derive` should call this function and just add the part where leading zeros are removed.